server/router.py

```diff
@@ -90,17 +124,16 @@ async def parse_directory(
                 project_manager.update_project_status(project_id, ProjectStatusEnum.ERROR)
                 message = "Repository doesn't consist of a language currently supported."
         else:
-            dir_details = project_details["directory"]
-            project_id = project_details["id"]
-            if is_deleted:
+            dir_details = project_details.directory
+            project_id = project_details.id
+            if project_deleted:
                 message = project_manager.restore_project(project_id, user_id)
             else:
                 message = "The project has been re-parsed successfully"
-            if GithubService.check_is_commit_added(repo, project_details, branch_name):
-                reparse_cleanup(project_details, user_id)
-                dir_details, project_id, should_parse_repo = setup_project_directory(owner, repo_name,
-                                                                  branch_name, app_auth, repo, user_id,
-                                                                  project_id)
+            if repo_details.repo_path:
```

Comment:
should this if else logic be one level deeper

Reply:
No this seems correct to me.